pkg/config/remote_config.go

```diff
@@ -0,0 +1,127 @@
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/prometheus/common/config"
+)
+
+// supported remote config provider schemes
+const (
+	HTTP  = "http"
+	HTTPS = "https"
+	// TODO: add s3, gcs, blob, and git providers backed by go-fsimpl
+)
+
+// RemoteOpts struct contains agent remote config options
+type RemoteOpts struct {
+	ExpandEnv bool
+
+	URL              *url.URL
+	HTTPClientConfig *config.HTTPClientConfig
+}
+
+// RemoteProvider ...
+type RemoteProvider interface {
+	Retrieve() (*Config, error)
+}
+
+// RemoteConfig ...
+type RemoteConfig struct {
+	RemoteProvider
+}
```

Comment:
I'm not sure I understand why we have a struct that only embeds the interface? IMO, just return the interface from `NewRemoteConfig`

Reply:
I just did this for flexibility so if necessary additional values could be carried with this struct instead of just on the underlying `RemoteProvider` implementations.

Reply:
YAGNI :) I'd say keep it as an interface for now.

Reply:
ok -- switching to an interface is fine and it's easy to switch back if need be. it doesn't change the semantics for the caller. If we do end up with additional providers, conf.d style support, etc. I can see us wanting to carry other values here potentially.